src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -79,6 +81,13 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 				this.refreshLayouts();
 			} else if (e.affectsConfiguration('workbench.colorCustomizations')) {
 				this._refreshStyles(true);
+			} else if (e.affectsConfiguration(TerminalSettingId.ShellIntegrationDecorationsEnabled) || e.affectsConfiguration(TerminalSettingId.ShellIntegrationEnabled)) {
+				const enabledDecorations = this._configurationService.getValue(TerminalSettingId.ShellIntegrationDecorationsEnabled) && this._configurationService.getValue(TerminalSettingId.ShellIntegrationEnabled);
```

Comment:
This should be:

```suggestion
				const enabledDecorations = this._configurationService.getValue(TerminalSettingId.ShellIntegrationDecorationsEnabled)
```

`TerminalSettingId.ShellIntegrationEnabled` is a little misleading in the name but it's really "shell integration auto injection enabled", we recommend users disable `TerminalSettingId.ShellIntegrationEnabled` when they manually install shell integration.